### PR TITLE
games-roguelike/cataclysm-dda: added a dev branch USE

### PR DESCRIPTION
**Pull request by l29ah**
as announced in https://github.com/CleverRaven/Cataclysm-DDA/issues/36949#issuecomment-595704738

### CODE REVIEW

**Comment by leycec** [comment]
**Cataclysmic.** Thanks for the excellent wintertime volunteerism – as always, @l29ah.

Ordinarily, I'd prefer to unconditionally switch the `EGIT_BRANCH` from `master` to `dev` rather than add another USE flag. That said, @ZhilkinSerg's commentary *does* suggest that the new `dev` branch might just be a temporary placeholder that will go away after the `0.E` release. If so, then adding another USE flag is actually the Right Thing To Do.™

In short, time will tell. I'm delighted to merge this as is and grateful for your keen eye. Let's keep an eye on upstream, though. If unstable feature development ends up permanently moving over to the new `dev` branch, we'll probably want to:

1. Revert this commit.
1. Unconditionally switch the `EGIT_BRANCH` from `master` to `dev`.

Thanks again, @l29ah. You're awesomeness!

**Comment by KorGgenT** [comment]
as a developer of C:DDA i'll confirm that it is temporary and the `dev` branch will be merged to `master` after the 0.E stable release

**Comment by ZhilkinSerg** [comment]
FYI: We started to make hotfix releases for stable versions:

- https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/0.E-1
- https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/0.E-2

It is not necessary that each and every stable release in the future will have such hotfix, but it is highly likely.
